source/Mlos.Agent.Server/MlosAgentServer.cs

```diff
@@ -57,21 +60,30 @@ public static void Main(string[] args)
         {
             string executableFilePath = null;
             Uri optimizerAddressUri = null;
+            IEnumerable<string> extraArgs = null;
 
-            var cliOptsParseResult = CommandLine.Parser.Default.ParseArguments<CliOptions>(args)
+            var cliOptsParser = new Parser(with => with.HelpWriter = null);
+            var cliOptsParseResult = cliOptsParser.ParseArguments<CliOptions>(args)
                 .WithParsed(parsedOptions =>
                 {
                     executableFilePath = parsedOptions.Executable;
                     optimizerAddressUri = parsedOptions.OptimizerUri;
+                    extraArgs = parsedOptions.ExtraArgs;
                 });
             if (cliOptsParseResult.Tag == ParserResultType.NotParsed)
             {
-                // CommandLine already prints the help text for us in this case.
-                //
-                Console.Error.WriteLine("Failed to parse command line options.");
-                Environment.Exit(1);
+                cliOptsParseResult.WithNotParsed(errs => ShowUsageHelp(
+                    cliOptsParseResult,
+                    errors: errs,
+                    msg: "Failed to parse command line options."));
+            }
+            else if (extraArgs != null && extraArgs.Any())
+            {
+                ShowUsageHelp(cliOptsParseResult, msg: "Extra arguments found.");

```

Comment:
Should this be 'Unknown' arguments instead of 'Extra'?

Reply:
CommandLineParser has a concept of "unknown" _options_ internally, but it currently passes unknown _arguments_ as is (in case another framework wants to handle them).  In this sense, I think of them as they're "extra" arguments that are passed to us.  We then interpret them all as "unknown" since we don't have a second CLI framework to handle them.

Anyways, that's my naming logic.  Happy to change it if you want though.

Reply:
Sorry, were you asking about the output message or the variable name?

Reply:
Changed.  Let me know if you have any other feedback on it.